src/openrct2-ui/windows/TrackDesignPlace.cpp

```diff
@@ -240,6 +241,30 @@ static void window_track_place_update(rct_window* w)
             window_close(w);
 }
 
+static GameActionResult::Ptr FindValidTrackDesignPlaceHeight(CoordsXYZ& loc, uint32_t flags)
+{
+    for (int32_t i = 0; i < 7; i++)
+    {
+        auto tdAction = TrackDesignAction(CoordsXYZD{ loc.x, loc.y, loc.z, _currentTrackPieceDirection }, *_trackDesign);
+        tdAction.SetFlags(flags);
+        auto res = GameActions::Query(&tdAction);
+
+        // If successful dont keep trying.
+        // If failure due to no money then increasing height only makes problem worse
+        if (res->Error == GA_ERROR::OK || res->Error == GA_ERROR::INSUFFICIENT_FUNDS)
+        {
+            return res;
+        }
+        // Return the last attempts error up the chain
+        if (i == 6)
+        {
+            return res;
+        }
+        loc.z += 8;
+    }
+    return nullptr;
```

Comment:
This code will always return `res`, so you might just move the comment to just before the last return, change it to return `res` and remove this special handling for `i == 6`, because if you increment `loc.z` on the `for` like I proposed, it will do the same number of increments (https://godbolt.org/z/M-_yQ2)